client/my-sites/stats/activity-log/threat-alert.jsx

```diff
@@ -35,38 +29,58 @@ export class ThreatAlert extends Component {
 						<div className="activity-log__threat-alert-header">
 							<div>
 								<span className="activity-log__threat-alert-title">
-									{ translate( 'Thread found' ) }
+									{ threat.diff && translate( 'Threat found in core WordPress file' ) }
+									{ threat.extension &&
+										threat.extension.type === 'plugin' &&
+										translate( 'Threat found in your {{s}}%(slug)s{{/s}} Plugin', {
+											args: {
+												slug: threat.extension.slug,
+											},
+											components: {
+												s: <span className="activity-log__threat-alert-title-slug" />,
+											},
+										} ) }
+									{ threat.extension &&
+										threat.extension.type === 'theme' &&
+										translate( 'Threat found in your {{s}}%(slug)s{{/s}} Theme', {
+											args: {
+												slug: threat.extension.slug,
+											},
+											components: {
+												s: <span className="activity-log__threat-alert-title-slug" />,
+											},
+										} ) }
+									{ ! threat.extension && ! threat.diff && translate( 'Threat found' ) }
 								</span>
-								{ threat.files && (
-									<TimeSince
-										className="activity-log__threat-alert-time-since"
-										date={ earliestDetection }
-										dateFormat="ll"
-									/>
-								) }
+								<TimeSince
+									className="activity-log__threat-alert-time-since"
+									date={ threat.firstDetected }
+									dateFormat="ll"
+								/>
 							</div>
 							<span>{ translate( 'Malware code detected' ) }</span>
 						</div>
 					</Fragment>
 				}
 			>
-				{ threat.description }
-				{ threat.files &&
-					threat.files.map( file => (
-						<Fragment key={ file.name }>
-							<div className="activity-log__threat-alert-file-header">
-								<span className="activity-log__threat-alert-filename">{ file.name }</span>
-								<TimeSince
-									className="activity-log__threat-alert-time-since"
-									date={ file.firstDetected }
-									dateFormat="ll"
-								/>
-							</div>
-							<p className="activity-log__threat-alert-dirpath">{ file.dirpath }</p>
-							{ file.context && <MarkedLines context={ file.context } /> }
-							{ file.diff && <DiffViewer diff={ file.diff } /> }
-						</Fragment>
-					) ) }
+				{ ! threat.filename && (
+					<p className="activity-log__threat-alert-signature">{ threat.signature }</p>
```

Comment:
Maybe we should ensure that `threat.signature` exists before using it.

Reply:
it's required to exist and we'll make sure it doesn't enter Calypso unless it does